consul/tests/test_unit.py

```diff
@@ -458,6 +458,41 @@ def test_network_latency_checks(aggregator):
     assert 0.26577747932995816 == node[0][2]
 
 
+@pytest.mark.parametrize(
+    'use_node_id_as_hostname, expected_hostname',
+    [
+        ('false', ['']),
+        ('true', ['host-1', 'host-2']),
+    ],
+)
+def test_network_latency_node_name(aggregator, use_node_id_as_hostname, expected_hostname):
+    consul_mocks.MOCK_CONFIG_NETWORK_LATENCY_CHECKS['use_node_id_as_hostname'] = use_node_id_as_hostname
+
+    consul_check = ConsulCheck(common.CHECK_NAME, {}, [consul_mocks.MOCK_CONFIG_NETWORK_LATENCY_CHECKS])
+    consul_check.instance.update({'use_node_id_as_hostname': True})
+    consul_mocks.mock_check(consul_check, consul_mocks._get_consul_mocks())
+
+    # We start out as the leader, and stay that way
+    consul_check._last_known_leader = consul_mocks.mock_get_cluster_leader_A()
+
+    consul_check.check(None)
+
+    metrics = [
+        'consul.net.node.latency.max',
+        'consul.net.node.latency.median',
+        'consul.net.node.latency.min',
+        'consul.net.node.latency.p25',
+        'consul.net.node.latency.p75',
+        'consul.net.node.latency.p90',
+        'consul.net.node.latency.p95',
+        'consul.net.node.latency.p99',
+    ]
+
+    for m in metrics:
+        for host in expected_hostname:
+            aggregator.assert_metric(m, hostname=host)
```

Comment:
Since you also added the feature to have a new tag collected, can you also test that here?